tests/optional_kernel_features/kernel_features_mismatched_nd_range_exception.cpp

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides tests for the exception that is thrown by
+//  [[sycl::reqd_work_group_size]] attribute when use nd_range of wrong size in
+//  kernel.
+//
+*******************************************************************************/
+
+#include "../common/common.h"
+#include "catch2/catch_template_test_macros.hpp"
+
+namespace kernel_features_mismatched_nd_range_exception {
+using namespace sycl_cts;
+
+struct kernel_separate_lambda;
+struct kernel_functor;
+struct kernel_submission_call;
+
+// Define required size of work group for attribute
+constexpr int testing_wg_size = 16;
+
+class Functor {
+ public:
+  [[sycl::reqd_work_group_size(testing_wg_size)]] void operator()(
+      sycl::nd_item<1>) const {}
+};
+
+TEST_CASE(
+    "Kernel features. Exceptions throwns by [[reqd_work_group_size()]] with "
+    "mismatched nd_range",
+    "[kernel_features]") {
+  auto queue = util::get_cts_object::queue();
+  const size_t max_wg_size =
+      queue.get_device().get_info<sycl::info::device::max_work_group_size>();
+  if (max_wg_size < testing_wg_size) {
+    WARN("Device supported work group size too small. Skipping..");
+    return;
+  }
+
+  const bool is_exception_expected = true;
+  sycl::errc errc_expected = sycl::errc::nd_range;
+
+  const auto separate_lambda = [=](sycl::nd_item<1>)
+      [[sycl::reqd_work_group_size(testing_wg_size)]]{};
+
+  // Create nd_range that have to cause an exception
+  sycl::nd_range<1> mismatched_nd_rage(sycl::range(max_wg_size + 1),
+                                       sycl::range(max_wg_size + 1));
+
+  {
+    INFO("Task as separate lambda");
+    bool is_exception_thrown = false;
+    try {
+      queue
+          .submit([&](sycl::handler& cgh) {
+            cgh.parallel_for<kernel_separate_lambda>(mismatched_nd_rage,
+                                                     separate_lambda);
+          })
+          .wait_and_throw();
+    } catch (const sycl::exception& e) {
+      is_exception_thrown = true;
+      INFO("Error code check");
+      CHECK(e.code() == errc_expected);
```

Comment:
These checks could be replaced with `CHECK_THROWS_MATCHES` introduced by @yuriykoch in #274!

Reply:
Looks like #274 is still not merged. 
I planned to patch these tests after `SKIP` will be introduced in `catch2` https://github.com/catchorg/Catch2/pull/2360. I think Yuriy's patch will be merged by that time so I will replace this part of code with `CHECK_THROWS_MATCHES`.

Reply:
Okay sure. `SKIP` will still take a while though, I'm afraid.